lib/skate_web/controllers/detours_controller.ex

```diff
@@ -4,10 +4,28 @@ defmodule SkateWeb.DetoursController do
   alias Skate.OpenRouteServiceAPI
   use SkateWeb, :controller
 
+  alias Skate.Detours.Detours
   alias Skate.Detours.MissedStops
   alias Skate.Detours.RouteSegments
+  alias SkateWeb.AuthManager
   alias Util.Location
 
+  @spec update_snapshot(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def update_snapshot(conn, %{"snapshot" => %{"context" => context} = snapshot}) do
+    uuid = Map.get(context, "uuid")
+
+    %{id: user_id} = AuthManager.Plug.current_resource(conn)
+
+    {:ok, returned_detour} =
+      Detours.update_or_create_detour_for_user(user_id, uuid, %{
+        state: snapshot
+      })
+
+    returned_uuid = Map.get(returned_detour, :id)
```

Comment:
suggestion: would this work?
```suggestion
    {:ok, %Skate.Detours.Db.Detour{id: returned_uuid} = returned_detour} =
      Detours.update_or_create_detour_for_user(user_id, uuid, %{
        state: snapshot
      })
```

Reply:
You bet!